### ISSUE TITLE
Issue using certificate map

### ISSUE THREAD

**Issue by lbordowitz** [terraform-google-modules/terraform-google-lb-http#372]
### TL;DR

I'm having a lot of trouble using the `certificate_map` variable. Could you make an example of it being used?

### Additional information

I've tried with `use_ssl_certificates=true` and `use_ssl_certificates=false`. If it's `false`, the `private_key` and `certificate` are required; these overrule any certificate map setting. If it's `true`, it requires `ssl_certificates` be nonempty. However, `ssl_certificates` will not accept a `google_certificate_manager_certificate`; the last part of the resource is the only part that's referenced, and the resource url is rewritten as a classic certificate instead of being a "Certificate Manager" certificate. So, I don't know how to actually use the `certificate_map` variable that's provided. In fact, I don't think I've ever seen a "Certificate Manager" certificate successfully used with the https_proxy resource.

**Comment by imrannayer** [comment]
In order to pass certificate_map, `use_ssl_certificates` needs to be `false` and `ssl_certificates` will be `[]`. Module is not able to handle this scenario. Changing this issue as bug.

**Comment by imrannayer** [comment]
@lbordowitz #382 will fix the issue. I had to make changes to some parameters as old parameters were too limited and was causing certificate map issues.
